stytch/shared/rbac_local.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+from typing import List
+
+import pydantic
+
+from stytch.b2b.models.rbac import Policy
+
+
+class TenancyError(ValueError):
+    def __init__(self, subject_org_id: str, request_org_id: str) -> None:
+        self.subject_org_id = subject_org_id
+        self.request_org_id = request_org_id
+
+    def __str__(self):
+        return (
+            f"Subject organization ID {self.subject_org_id} does not "
+            f"match request organization ID {self.request_org_id}"
+        )
+
+
+class PermissionError(ValueError):
+    def __init__(self, authz_request: AuthZRequest) -> None:
+        self.authz_request = authz_request
+
+    def __str__(self):
+        return f"Permission denied for {self.authz_request}"
+
+
+class AuthZRequest(pydantic.BaseModel):
+    organization_id: str
+    resource_id: str
+    action: str
+
+
+def perform_authorization_check(
```

Comment:
Could we get a set of testcases for this function? There should be some we can copy over directly from the Node PR